src/backends/github/implementation.js

```diff
@@ -139,8 +139,10 @@ export default class GitHub {
       let url = `https://raw.githubusercontent.com/${repo}/${this.branch}${path}`;
 
       // Assets uploaded to private repos will need valid access tokens.
-      const isPrivateRepo = await this.api.isPrivateRepo();
-      if (isPrivateRepo) {
+      if (!this.isPrivateRepo) {
+        this.isPrivateRepo = await this.api.checkPrivateRepo(`${this.api_root}/repos/${repo}`);
```

Comment:
The value is cached here since GitHub limits unauthenticated API requests. The only downside is that you need to reload the CMS page if you change the repo from public to private.

Reply:
I think that's a fine compromise.